r2-shared/src/main/java/org/readium/r2/shared/publication/services/search/StringSearchService.kt

```diff
@@ -103,6 +107,8 @@ class StringSearchService(
                 }
 
                 val locators = findLocators(index, link, text)
+                resultCount = (resultCount ?: 0) + locators.count()
```

Comment:
```suggestion
                resultCount = resultCount + locators.count()
```

Might not need null check at all if type is converted to not nullable

Reply:
See https://github.com/readium/r2-shared-kotlin/pull/154#discussion_r635273765